src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -776,12 +790,43 @@ export class SearchResult extends Disposable {
 		this._register(this.modelService.onModelAdded(model => this.onModelAdded(model)));
 
 		this._register(this.onChange(e => {
+			this._onRefreshSearchResult.fire(e);
 			if (e.removed) {
 				this._isDirty = !this.isEmpty();
 			}
 		}));
 	}
 
+	async performMassReplace(elementsToReplace: RenderableMatch[]) {
+		this._onRefreshSearchResult.pause();
+		await Promise.all(elementsToReplace.map(async (elem) => {
+			const parent = elem.parent();
+
+			if ((parent instanceof FolderMatch || parent instanceof FileMatch) && arrayContainsElementOrParent(parent, elementsToReplace)) {
+				// skip any children who have parents in the array
+				return;
+			}
+
+			if (elem instanceof FileMatch) {
+				elem.parent().replace(elem);
```

Comment:
You can use the `parent` from above.

Reply:
typescript seems to not like that too much

![image](https://user-images.githubusercontent.com/31675041/184255364-73d643ee-e81b-45f5-8cc1-cc46caa46bfb.png)

Reply:
^ probably because it's able to evaluate the exact type of the parent when it's evaluated after the explicit type check